pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -5549,3 +5549,152 @@ func TestContinueOnError(t *testing.T) {
 	assert.Equal(t, resource.URN("urn:pulumi:test::test::pulumi:providers:pkgB::default"), snap.Resources[2].URN)
 	assert.Equal(t, resource.URN("urn:pulumi:test::test::pkgB:m:typB::failing"), snap.Resources[3].URN)
 }
+
+func TestUpContinueOnErrorCreate(t *testing.T) {
+	t.Parallel()
+
+	loaders := []*deploytest.ProviderLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{}, nil
+		}, deploytest.WithoutGrpc),
+		deploytest.NewProviderLoader("pkgB", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				CreateF: func(urn resource.URN, news resource.PropertyMap, timeout float64,
+					preview bool,
+				) (resource.ID, resource.PropertyMap, resource.Status, error) {
+					return "", nil, resource.StatusOK, errors.New("intentionally failed create")
+				},
+			}, nil
+		}, deploytest.WithoutGrpc),
+	}
+
+	programF := deploytest.NewLanguageRuntimeF(func(_ plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
+		_, _, _, _, err := monitor.RegisterResource("pkgB:m:typB", "failing", true, deploytest.ResourceOptions{})
+		assert.NoError(t, err)
```

Comment:
assert.NoError? Surely this _does_ error now?

Reply:
It doesn't error during the `RegisterResource` call.  The errors are captured in `OnResourceStepPost`, and reported back to the user that way.  An earlier iteration of this that I experimented with did error on `RegisterResource`, but that means the error ends up getting reported to the user twice.  If I understand correctly, `RegisterResource` only errors when there is an error the engine captures, e.g. invalid URNs, etc.

Note that not erroring here when the provider returns an error is also the current behaviour without any `ContinueOnError` flag.

Reply:
> Note that not erroring here when the provider returns an error is also the current behaviour without any

Really... hmm that's worrying. The SDK then thinks this resource is OK and tries to access the outputs but the outputs aren't going to exist?

Reply:
Oh wait, I think I was confusing something here. we're not returning an error here, but we're also not returning any response.  The whole `programF` will just hang in the non ContinueOnError case, and we're exiting because the context is canceled.  But we're never even hitting this assert if `ContinueOnError` is false.